### DISCUSSION TITLE
Next.config.js - redirects... questions...

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37649] [Help] [answered]
Ok.. first... what is this code?  why is there?  and what is it exactly saying ?

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
}

module.exports = nextConfig
```

I see something with /** ....  do I need to remove that?

But on the other hand what I want to do is a re-direction of any non-existing page to go to the home page...   maybe something like this???

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,


async redirects() {
    return [
      {
        source: '/*',
        destination: '/',
        permanent: true,
      },
    ]
  },

}

```

oooohhhhh BTW.....  Vercel gave me an https  and I have a huge load of old web addresses with the old http.....   would that code above re-direct whatever old address I have to the new home page?

**Comment by icyJoseph** [comment, marked as answer]
The code underneath provides TypeScript auto-completion, so that you can write valid `next.config.js` objects, inside of a JavaScript file. There's other techniques to do, but this one is quite clean and neat! Though it only works if the types are provided by the package, roughly speaking.

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
}

module.exports = nextConfig
```

Sorta like this (it tells me possible keys, and even their value type):

<img width="614" alt="Screenshot 2022-06-12 at 21 29 36" src="https://user-images.githubusercontent.com/21013447/173250094-69da72db-7a3d-48f0-863a-d1b9e7beb30c.png">

It is possible thanks to : `/** @type {import('next').NextConfig} */` right before the assignment to `nextConfig`

### Issue at hand

I don't know if there's a good way to redirect in this case, through `next.config.js` anyway. 

But here's my take on it...

#### next.config.js

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      afterFiles: [{ source: "/:path*", destination: "/_404/:path*" }],
    };
  },
};

module.exports = nextConfig;
```

#### pages/_404/[...path].tsx (or just .js)

```tsx
// renders nada
export const Custom404 = () => <div></div>;

export const getServerSideProps = () => {
  return { redirect: { destination: "/", permanent: false } }; 
};

export default Custom404;
```

#### Explanation

> While developing, use permanent true with discretion, look at what this person went through https://github.com/vercel/next.js/discussions/37580

When a request arrives, after files are matched, and nothing is found, (including the public folder), then we rewrite to `/_404/[...path]`, where a `getServerSideProps` function does a redirect on all requests!

If people try to access `/_404/[...path]`, they'll just end up on the landing page as well!

---

And I do not really understand what you mean with the `http` vs `https` question.

**Comment by Elindo586** [comment]
you got skills..
